internal/ent/hooks/event.go

```diff
@@ -0,0 +1,249 @@
+	orgsettingID, exists := props["ID"]
+	if !exists {
+		log.Info().Msg("organizationSetting ID not found in event properties")
+		return nil
+	}
+
+	// if stripe ID is being added to the properties, we enumerate through any other fields in the update and ensure they match
+	stripeID, exists := props["stripe_id"]
+	if exists && stripeID != "" {
+		stripes := stripeID.(string)
+
+		stripecustomer, err := event.Client().(*entgen.Client).EntitlementManager.Client.Customers.Get(stripes, nil)
+		if err != nil {
+			log.Err(err).Msg("Failed to retrieve Stripe customer by ID")
+			return err
+		}
+
+		billingEmail, exists := props["billing_email"]
+		if exists && billingEmail != "" {
+			email := billingEmail.(string)
+			if stripecustomer.Email != email {
+				_, err := event.Client().(*entgen.Client).EntitlementManager.Client.Customers.Update(stripes, &stripe.CustomerParams{
+					Email: &email,
+				})
+				if err != nil {
+					log.Err(err).Msg("failed to update Stripe customer email")
+					return err
+				}
+			}
+		}
+
+		billingPhone, exists := props["billing_phone"]
+		if exists && billingPhone != "" {
+			phone := billingPhone.(string)
+			if stripecustomer.Phone != phone {
+				_, err := event.Client().(*entgen.Client).EntitlementManager.Client.Customers.Update(stripes, &stripe.CustomerParams{
+					Phone: &phone,
+				})
+				if err != nil {
+					log.Err(err).Msg("failed to update Stripe customer phone")
+					return err
+				}
+			}
+		}
+
+		// TODO: split out address fields in ent schema
+		billingAddress, exists := props["billing_address"]
+		if exists && billingAddress != "" {
+			address := billingAddress.(string)
+			if stripecustomer.Address.Line1 != address {
+				_, err := event.Client().(*entgen.Client).EntitlementManager.Client.Customers.Update(stripes, &stripe.CustomerParams{
+					Address: &stripe.AddressParams{
+						Line1: &address,
+					},
+				})
+				if err != nil {
+					log.Err(err).Msg("failed to update Stripe customer address")
+					return err
+				}
+			}
+		}
+	}
+
+	billingEmail, exists := props["billing_email"]
+	if exists && billingEmail != "" {
+		email := billingEmail.(string)
+
+		i := event.Client().(*entgen.Client).EntitlementManager.Client.Customers.List(&stripe.CustomerListParams{Email: &email})
+
+		if !i.Next() {
+			customer, err := event.Client().(*entgen.Client).EntitlementManager.Client.Customers.New(&stripe.CustomerParams{Email: &email})
+			if err != nil {
+				log.Err(err).Msg("Failed to create Stripe customer")
+				return err
+			}
+
+			log.Info().Msgf("Created Stripe customer with ID: %s", customer.ID)
+
+			if err := updateOrganizationSettingWithCustomerID(event.Context(), orgsettingID.(string), customer.ID, event.Client()); err != nil {
+				log.Err(err).Msg("Failed to update OrganizationSetting with Stripe customer ID")
+				return err
+			}
+
+			log.Info().Msgf("Updated OrganizationSetting with Stripe customer ID: %s", customer.ID)
```

Comment:
Most of these `Info` levels should probably be `Debug` because they shouldn't be needed in most cases because they are success